src/components/Card/index.js

```diff
@@ -1,14 +1,24 @@
 import React from "react";
 import { Card } from "react-materialize";
-const GenericCard = ({ date, price, start, dest }) => {
-  return (
-    <Card>
-      {date} <span style={{ float: "right" }}> {price} </span>
-      <h3>
-        {" "}
-        {start} to {dest}{" "}
-      </h3>
-    </Card>
-  );
-};
+class GenericCard extends React.Component {
+  constructor(props) {
+    super(props);
+    this.openModal = this.openModal.bind(this);
+  }
+  openModal() {
```

Comment:
Passing down getByID() and the id as props; putting functionality inside card. I prefer this to the alternative of rebinding functions inside the container, i.e. when we map trips in dashboard.